edb/pgsql/delta.py

```diff
@@ -890,34 +890,113 @@ def compile_edgeql_function(self, func: s_funcs.Function, schema, context):
 
         return self.make_function(func, sql_text, schema)
 
+    def sql_rval_consistency_check(
+        self,
+        cobj: s_funcs.CallableObject,
+        expr: str,
+        schema: s_schema.Schema,
+    ) -> dbops.Command:
+        fname = cobj.get_verbosename(schema)
+        rtype = types.pg_type_from_object(
+            schema,
+            cobj.get_return_type(schema),
+            persistent_tuples=True,
+        )
+        rtype_desc = '.'.join(rtype)
+
+        f_test = textwrap.dedent(f'''\
```

Comment:
Add at least a quick comment here explaining what this is accomplishing.

If you can stomach it, it might also be worth explaining *how*/*why* this works since I certainly don't understand it why this version works and other variants don't.

Reply:
Done.